Internal Twig Updates

japaveh opened the pull request:
To be able to use the software with the latest version of Twig some small (BC) patches are needed.
Since this fork is closest to ZF3 I created a fork of this one.

Commit:
Move from Twig_Function to Twig_SimpleFunction

Twig_Function is deprecated, move to Twig_Simplefunction and slightly change the footprint of the call to compile()

Commit:
Update findTemplate()

FindTemlate now requires a second optional parameter to supress the throwing of the exception

An exception has been thrown during the rendering of a template ("Error: In ZfcTwig\Twig\StackLoader::findTemplate(), you must accept a second argument that when set to "false" returns "false" instead of throwing an exception. Not supporting this argument is deprecated since version 1.27.

kokspflanze commented on src/ZfcTwig/Twig/FallbackFunction.php:
```diff
@@ -5,7 +5,7 @@
 use Twig_Function;
```
you can also change `Twig_Function` to `Twig_SimpleFunction` so we can also remove the `\` before the `Twig_SimpleFunction` class

kokspflanze commented on src/ZfcTwig/Twig/StackLoader.php:
```diff
@@ -41,18 +41,24 @@ public function getDefaultSuffix()
         $defaultSuffix = $this->getDefaultSuffix();
-        if (pathinfo($name, PATHINFO_EXTENSION) != $defaultSuffix) {;
+        if (pathinfo($name, PATHINFO_EXTENSION) != $defaultSuffix) {
+            ;
```
can you please remove the `;` too

kokspflanze commented on src/ZfcTwig/Twig/StackLoader.php:
```diff
@@ -41,18 +41,24 @@ public function getDefaultSuffix()
         $name = preg_replace('#/{2,}#', '/', strtr($name, '\\', '/'));
 
         if (isset($this->cache[$name])) {
-            return $this->cache[$name];
```
i dont realy get it, why throw an `Exception`, if the catch match?

i think it should be like https://github.com/twigphp/Twig/blob/1.x/lib/Twig/Loader/Filesystem.php#L203-L213

kokspflanze reviewed (changes requested):
thanks, the important part is this https://github.com/kokspflanze/ZfcTwig/pull/3/files/1971d839036c08d3df497d35ac999763e9dc8a89#diff-1379b576bb8f6c4f41991ba2423983b6 can you give me a reason for this?

Commit:
Update StackLoader.php

Small update, to be in line with https://github.com/twigphp/Twig/blob/1.x/lib/Twig/Loader/Filesystem.php#L203-L213

Commit:
Update StackLoader.php

Added the correct @phpdoc tags

Commit:
Update FallbackFunction.php

Update the use statement
Added the @phpdoc statements

japaveh commented:
The strange exception was a copy-paste error from Twig, my bad. I fixed it now

kokspflanze commented:
thanks, i will do some checks and add soon a new release.
